File: python/10809.py
s = input()[:100]
r = ""
for k in range(ord("a"), ord("z") + 1):
    ol = len(s)
    ll = len(s.split(chr(k))[0])
    r += f"{ll if ol != ll else -1 } "
print(r)


# # s = input()[:100]
# r = ""
# for k in "abcdefghijklmnopqrstuvwxyz":
#     i = 0
#     found = False
#     for i, c in enumerate(s):
#         if c == k:
#             r += f"{i} "
#             found = True
#             break
#     if len(s) == 0 or not found:
#         r += f"-1 "
# print(r)

File: rust/src/bin/10809.rs
use std::io;

fn main() {
  let mut i = String::new();
  let s = String::from("abcdefghijklmnopqrstuvwxyz");

  io::stdin()
    .read_line(&mut i)
    .expect("input error");
  

  let positions: Vec<i32> = s.chars()
                .map(
                  |x| {
                    i.chars()
                    .enumerate()
                    .fold( -1, |acc, (index, c)| {
                        if acc == -1 && c == x {
                          index as i32
                        } else {
                          acc
                        }
                      }
                    )  
                  }
                ).collect();
  
  print!("{}", positions.iter().map(|x| x.to_string()).collect::<Vec<_>>().join(" "));

}